scripts/5-run_normalization.R

```diff
@@ -0,0 +1,175 @@
+         'deseq2', 'vsd', 'rld' or 'all")
+}
+#----------------------------------Load data-----------------------------------#
+# Read in a tsv file of data
+dataset <- readr::read_tsv(opt$data)
+
+# Separate genes from the numeric data
+genes <- dataset[, grep("gene", colnames(dataset), ignore.case = TRUE)]
+
+# Make dataset the data only
+dataset <- dataset[, -grep("gene", colnames(dataset), ignore.case = TRUE)]
+
+#-----------------------Run each algorithm in the list-------------------------#
+# For each algorithm, run through this loop
+for (algorithm in opt$algorithm) {
+    
+    # Print out progress message:
+    cat("\n \n Running:", algorithm, "...")
+    
+    # Run normalization algorithms
+    if (algorithm  == "scale"){
+        data.out <- as.data.frame(scale(dataset))
+        title <- "Scaled Expression"
+        
+    } else  if (algorithm == "log") {
+        data.out <- sign(dataset) * log2(1 + abs(dataset))
+        title <- "Log2 Expression"
+        
+    } else  if (algorithm == "voom") {
+        data.out <- as.data.frame(limma::voom(counts = dataset,
+                                              normalize.method = "quantile",
+                                              plot = FALSE)$E)
+        title <- "Log2 Expression (Voom)"
+        
+    } else if (algorithm == "tmm") {
+        data.dge <- edgeR::DGEList(counts = dataset)
+        data.dge <- edgeR::calcNormFactors(data.dge)
+        data.out <- as.data.frame(edgeR::cpm(data.dge, normalized.lib.sizes = TRUE,
+                                             log = TRUE))
+        title <- "Log2 Expression (TMM / edgeR)"
+        
+    } else if (algorithm == "deseq") {
+        data.colData <- data.frame(row.names = colnames(dataset))
+        data.dds <- DESeq2::DESeqDataSetFromMatrix(dataset, colData = data.colData,
+                                                   design = ~1)
+        data.dds <- DESeq2::estimateSizeFactors(data.dds)
+        data.out <- as.data.frame(DESeq2::counts(data.dds, normalized = TRUE))
+        data.out <- sign(data.out) * log2(1 + abs(data.out))
+        title <- "Log2 Expression (DEseq2)"
+        
+    } else if (algorithm == "vsd") {
+        data.colData <- data.frame(row.names = colnames(dataset))
+        data.dds <- DESeq2::DESeqDataSetFromMatrix(dataset, colData = data.colData,
+                                                   design = ~1)
+        data.out <- SummarizedExperiment::assay(
+                    DESeq2::varianceStabilizingTransformation(data.dds, 
+                                                              blind = TRUE))
+        title <- "Log2 Expression (varianceStabilizingTransformation - DEseq2)"
+        
+    }
+    # Calculate percent zeroes
+    perc.zero <- round(length(which(data.out == 0)) / length(as.matrix(data.out)), 3)
+    
+    # Build output file name:
+    output.file <- paste0(opt$output, "/", algorithm, "_", opt$label, ".tsv")
```

Comment:
Couple things here:
- You should use `file.path()` for the path separator bit.
- If I were to not specify a label, the output file name would be (for log): `log_.tsv`, is that correct? Seems kind of ugly.